### PR TITLE
fixes bug news url method

### PR DESCRIPTION
**Pull request by ghost**
I think an extra parentheses are needed to prevent errors when isset($sharableData['image']) is false.

A ? A : B ? B : C
If the result of A ? A : B is true we evaluate the 2nd B as if B were true, which is not always correct.

If we use the parentheses : A ? A : (B ? B : C) then if A is false, we evaluate the whole 2nd ternary expression.

### CODE REVIEW

**Review by shaoshiva — CHANGES_REQUESTED** [review]
A ternary in a ternary is just awful for readability and should be avoided, even with parenthesis.

I would have done this :
```
if (isset($sharableData['image'])) {
    $imgId = (int)$sharableData['image'];
} elseif (isset($post->medias->thumbnail)) {
    $imgId = $post->medias->thumbnail->url();
} else {
    $imgId = null;
}
```

**Comment by ghost** [comment]
And I think you are absolutly right @shaoshiva 
I have made a new commit, same thing but in an if condition.

**Review by shaoshiva — APPROVED** [review]

**Review by Meroje — APPROVED** [review]

File: classes/controller/front.ctrl.php [hunk 1]
```diff
@@ -449,7 +449,13 @@ public function display_item()
 
             $this->main_controller->addMeta('<meta property="og:url" content="'.$post->url().'" />');
 
-            $imgId = isset($sharableData['image']) ? (int) $sharableData['image'] : isset($post->medias->thumbnail) ? $post->medias->thumbnail->url() : null;
+            $imgId = null;
+            if(isset($sharableData['image'])) {
```
Comment (shaoshiva) [inline classes/controller/front.ctrl.php]: Space between `if` and parenthesis.

File: classes/controller/front.ctrl.php [hunk 2]
```diff
@@ -449,7 +449,13 @@ public function display_item()
...
+                $imgId = (int)$sharableData['image'];
+            } else if (isset($post->medias->thumbnail)) {
```
Comment (shaoshiva) [inline classes/controller/front.ctrl.php]: No space between `else` and `if`.

**Review by shaoshiva — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@shaoshiva check out the last commit !

**Review by shaoshiva — APPROVED** [review]

**Pull request merged**